pkg/controllers/pod/pod_controller.go

```diff
@@ -96,10 +175,10 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		}
 		_, err = r.SubnetPortService.CreateOrUpdateSubnetPort(pod, nsxSubnet, contextID, &pod.ObjectMeta.Labels)
 		if err != nil {
-			r.StatusUpdater.UpdateFail(ctx, pod, err, "", nil)
+			r.StatusUpdater.UpdateFail(ctx, pod, err, "", setPodReadyStatusFalse, nil)
```

Comment:
Do we plan to update Pod status for those 2 errors:
https://github.com/vmware-tanzu/nsx-operator/blob/fff9bc7fbe104fb9e42f3f2dc93ce7bca37ef20c/pkg/controllers/pod/pod_controller.go#L87-L91
https://github.com/vmware-tanzu/nsx-operator/blob/fff9bc7fbe104fb9e42f3f2dc93ce7bca37ef20c/pkg/controllers/pod/pod_controller.go#L93-L96

Reply:
I added handling for the error from `r.SubnetService.GetSubnetByPath(nsxSubnetPath) `
but for the error from get node ID for pod, since there are some expected errors in the controller start, I feel that we don't need to add this error in pod status message.